controllers/storagecluster/external_resources.go

```diff
@@ -230,9 +230,49 @@ func (r *StorageClusterReconciler) newExternalCephObjectStoreInstances(
 // ensureCreated ensures that requested resources for the external cluster
 // being created
 func (obj *ocsExternalResources) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) (reconcile.Result, error) {
-	if r.sameExternalSecretData(instance) {
-		return reconcile.Result{}, nil
+
+	if isExternalOCSProvider(instance) {
+
+		externalClusterClient, err := r.newExternalClusterClient(instance)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		r.externalClusterClient = externalClusterClient
+		defer r.externalClusterClient.Close()
+
+		if instance.Status.ExternalStorage.ConsumerID == "" {
+			res, err := r.onboardConsumer(instance)
+			if err != nil || !res.IsZero() {
+				return res, err
+			}
+		} else if instance.Spec.ExternalStorage.RequestedCapacity.Equal(instance.Status.ExternalStorage.GrantedCapacity) {
+			res, err := r.updateConsumerCapacity(instance)
+			if err != nil || !res.IsZero() {
+				return res, err
+			}
+		}
+
+		if externalOCSResources[instance.UID] == nil {
+			externalConfig, res, err := r.getExternalConfigFromProvider(instance)
+			if err != nil || !res.IsZero() {
+				return res, err
+			}
+			externalOCSResources[instance.UID] = externalConfig
+		}
```

Comment:
Can we please get an `externalClusterClient.close()` here to close early if everything succeeded?

Reply:
no that can not be done, as we need to handle a failure in b/w.

Reply:
made changes